tasks/human_eval_051.rs

```diff
@@ -9,7 +9,45 @@ use vstd::prelude::*;
 
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+// This spec function checks a character is vowel or not
+pub open spec fn is_vowel_spec(c: char) -> bool {
+    c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'A' || c == 'E' || c == 'I'
+        || c == 'O' || c == 'U'
+}
+
+// This auxilary exe function checks a character is vowel or not
+fn is_vowel(c: char) -> (is_vowel: bool)
+    ensures
+        is_vowel == is_vowel_spec(c),
+{
+    c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u' || c == 'A' || c == 'E' || c == 'I'
+        || c == 'O' || c == 'U'
+}
+
+// Implementation following the ground-truth
+// This function removes vowels from a given string
+fn remove_vowels(str: &[char]) -> (str_without_vowels: Vec<char>)
+    ensures
+        str_without_vowels@ == str@.filter(|x: char| !is_vowel_spec(x)),
+{
+    let ghost str_length = str.len();
+    let mut str_without_vowels: Vec<char> = Vec::new();
+    assert(str@.take(0int).filter(|x: char| !is_vowel_spec(x)) == Seq::<char>::empty());
+
+    for index in 0..str.len()
+        invariant
+            0 <= index <= str.len(),
```

Comment:
I think you get the invariant `0 <= index <= str.len()` for free from the `for` loop.  Indeed, it appears to [verify without it](https://play.verus-lang.org/?version=stable&mode=basic&edition=2021&gist=991f834329055db333ef9c8af8cd97c7).

Reply:
Thanks for the feedback.